harness/determined/pytorch/_data.py

```diff
@@ -245,13 +248,18 @@ def get_data_loader(
             batch_sampler, repeat=repeat, skip=skip, num_replicas=num_replicas, rank=rank
         )
 
-        # Try to no break any torch version as old as v1.0.
+        # Try to not break any torch version as old as v1.0.
```

Comment:
since we stopped supporting torch 1.7... remove lines 253–257?

Reply:
Done, I removed anything that references version<1.10 in this file. :)

Reply:
Wait, I know this is after-merge, but when did we stop supporting torch 1.7?

There's a difference between "our environments support 1.7" and "our APIs support 1.7".  I think, if torch is api-compatible to code written against 1.0, we should still be supporting those versions, at least at the API layer.

Reply:
we had a note in 0.22.0 saying that we’ve stopped supporting tf 1 and torch 1.7: https://docs.determined.ai/latest/release-notes.html#version-0-22-0

the note does not make distinction between environments and APIs, TF1 APIs will stop being supported quite explicitly. when we discussed it, I don't recall anyone coming in defence of torch 1.7. if anything, people are asking for newer pytorch versions all the time. so I think the users have been notified appropriately in this case.

Reply:
Ok, I think the note is wrong in its approach then.

Not supporting those versions in our environments? Sure.

Not supporting tf1 in the keras code we own?  Sure.  That support was expensive, and tf1 vs tf2 was a huge breaking change.

Not supporting Pytorch 1.0 in the pytorch code we own?  I don't agree with that.  Pytorch continues to be stable back to 1.0, and so I think the (very small) vendored pytorch code we have should also have that kind of stability.

It costs nothing to just leave it alone, and it might help somebody someday.

It might never help anybody, too.

I don't see where we draw the line on "how recent of pytorch do we support" with the system you are suggesting.  I don't want our oldest environments to be the oldest thing we support.  I'd like us to update environments much more frequently, but to continue to work with old pytorch just like pytorch itself does.  I'd like more customers with those situations to feel safe owning their own custom images and expecting them to work, because that decreases our environments burden.

But tying our pytorch api support to the pytorch environments we ship... that makes changing environments more expensive and more impactful to customers, and I think that's a bad idea.